lib/gems/pending/util/mount/miq_generic_mount_session.rb

```diff
@@ -34,7 +34,9 @@ def self.runcmd(cmd_str)
     # If sudo is required, ensure you have /etc/sudoers.d/miq
     # Cmnd_Alias MOUNTALL = /bin/mount, /bin/umount
     # %wheel ALL = NOPASSWD: MOUNTALL
-    rv = `sudo #{cmd_str} 2>&1` if rv.include?("mount: only root can do that")
+    if $CHILD_STATUS.exitstatus && cmd_str =~ /^(mount|umount) /
```

Comment:
This is still not correct, in ruby all integers are truthy.

I would say this should use either `!$CHILD_STATUS.success?` (which returns a `true` or `false` value) or check explicitly for exit code 1 which is defined as `incorrect invocation or permissions` in the `mount (8)` man pages.